compat/src/suspense.js

```diff
@@ -104,7 +104,14 @@ Suspense.prototype._childDidSuspend = function(promise, suspendingComponent) {
 		}
 	};
 
-	if (!c._pendingSuspensionCount++) {
+	/**
+	 * We do not set `suspended: true` during hydration because we want the actual markup
+	 * to remain on screen and hydrate it when the suspense actually gets resolved.
+	 * While in non-hydration cases the usual fallback -> component flow would occour.
+	 */
+	const vnode = c._vnode;
```

Comment:
I still need to run this code to verify my understanding, but my first reading through makes me think this may need to be `suspendingComponent._vnode` since we set `vnode._hydrating` on the vnode that threw the error (diff/index.js:275), not on `Suspense._vnode` (though we could in Suspense's `_catchError` implementation at the top of this file).

Reply:
You may be right. I need to clarify which vnode(s) we want to set the `_hydrating` bit on. My preference would be to set _hydrating on the vnode for the component that _caught/handled_ the suspend, since technically we bail out of the tree at that point and must resume from that point.

Reply:
Ah - yeah currently we're setting it on both the bailed-out VNode _and_ the VNode for the component that handled the bailout:
https://github.com/preactjs/preact/pull/2754/files#diff-b5e3843a998fbd0144a6ac5b1b7536a4R33

Reply:
Yea, that makes sense to me (setting in on the place in the tree where we bail out and resume). Though I think when using compat's Suspense, the `_catchError` code isn't run since `Suspense` ships it's own `_catchError` implementation that doesn't have these changes to set hydrating, but that'd be trivial to add.

Looking back at it, I'm don't remember why we didn't just use `componentDidCatch` to implement Suspense... Maybe we needed something like your `handled` logic but no one tried that at the time?

Reply:
I think the reason for not using `componentDidCatch` was that some users were confused why the received a `Promise` in their error handlers instead of an `Error` object. We originally implemented it that way iirc, but backpedaled.